lua/nui/tree/init.lua

```diff
@@ -340,11 +340,22 @@ function Tree:add_node(node, parent_id)
   self:_add_nodes({ node }, parent_node)
 end
 
+local function remove_node(tree, node_id)
+  local node = tree.nodes.by_id[node_id]
+  if node:has_children() then
+    for _, child_id in ipairs(node._child_ids) do
+      -- We might want to store the nodes and return them with the node itself?
+      tree.nodes.by_id[child_id] = nil
```

Comment:
Umm... the previous code was right. Child nodes can also have children. For example, in `neo-tree` folders can have sub-folders of arbitrary depth.

Reply:
> Also I pushed one more quick change to remove the recursion as it doesn't need to be there.

I think we do need the recursion.

Reply:
Oh ya, I kinda forgot about that. Whoops!